packages/ag-charts-community/src/chart/axis/axisUtil.ts

```diff
@@ -212,3 +219,61 @@ export function resetAxisLineSelectionFn() {
         return { ...datum };
     };
 }
+
+const CARTESIAN_AXIS_POSITIONS: AgCartesianAxisPosition[] = ['top', 'right', 'bottom', 'left'];
+const CARTESIAN_AXIS_TYPES: AgCartesianAxisType[] = ['category', 'grouped-category', 'number', 'log', 'time'];
+
+function hasCartesianAxisPosition(axis: ChartAxis): axis is ChartAxis & { position: AgCartesianAxisPosition } {
+    const allowedTypes: string[] = CARTESIAN_AXIS_TYPES;
+    return allowedTypes.includes(axis.type);
+}
+
+function isCartesianAxisOptions(options: AgBaseAxisOptions): options is AgBaseCartesianAxisOptions {
+    const allowedTypes: string[] = CARTESIAN_AXIS_TYPES;
+    return allowedTypes.includes(options.type);
+}
+
+function isAxisPosition(position: unknown): position is AgCartesianAxisPosition {
+    const allowedPositions: string[] = CARTESIAN_AXIS_POSITIONS;
+    return typeof position === 'string' && allowedPositions.includes(position);
+}
+
+// If axis[].position, then we cannot always default to the same value. We need
+// to default to an 'untaken' position (see AG-9963 for more info).
+export class AxisPositionGuesser {
```

Comment:
Hmmm... `axisUtil.ts` is really related to the axis implementation - I would keep this 'validation'-esque behaviour somewhere closer to the mapping/validation phase of processing. perhaps in `charts/mapping/prepareAxis.ts` similar to `prepareSeries.ts`?

Reply:
No objections. I'll move these functions.

Reply:
Done.